test/Sema/diag_deprecated_string_interpolation.swift

```diff
@@ -0,0 +1,24 @@
+// RUN: %target-swift-frontend -swift-version 5 -typecheck %s 2>&1 | %FileCheck %s
+
+extension DefaultStringInterpolation {
+    @available(*, deprecated) func appendInterpolation(deprecated: Int) {}
+}
+
+// Make sure diagnostics emitted via string interpolations have a reasonable source location
+
+_ = "\(deprecated: 42)"
+// CHECK: [[@LINE-1]]:7: warning: 'appendInterpolation(deprecated:)' is deprecated
```

Comment:
I thought at first that you had forgotten about `%target-typecheck-verify-swift`, but then I realized that it doesn't let you check column numbers, only line numbers. I've filed https://bugs.swift.org/browse/SR-11320 to improve that situation.